### PR TITLE
[wip] cmake modules

### PR DESCRIPTION
**Pull request by makslevental**

### CODE REVIEW

File: runtime/src/iree/hal/CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,27 @@ iree_cc_library(
     hal
   HDRS
     "api.h"
+    "allocator.h"
```
Comment (benvanik) [inline runtime/src/iree/hal/CMakeLists.txt:18]: not sure what you're doing here, but these are private files - _only_ api.h is public - so don't do this

File: runtime/src/iree/base/CMakeLists.txt [hunk 2]
```diff
@@ -45,11 +45,21 @@ iree_cc_library(
     core_headers
   HDRS
     "alignment.h"
+    "allocator.h"
```
Comment (benvanik) [inline runtime/src/iree/base/CMakeLists.txt:48]: same here - not sure what you're doing, but the core_headers explicitly _don't_ have these files

File: build_tools/cmake/CMakeLists.txt [hunk 3]
```diff
@@ -42,3 +46,17 @@ install(FILES
...
+  ${CMAKE_CURRENT_SOURCE_DIR}/external_cc_library.cmake
+  ${CMAKE_CURRENT_SOURCE_DIR}/flatbuffer_c_library.cmake
+)
+list(REMOVE_ITEM _iree_cmake_helpers iree_llvm.cmake)
+
+install(FILES
+   ${_iree_cmake_helpers}
+   DESTINATION ${_cmake_installdir}
+   COMPONENT IREECMakeExports
+)
```
Comment (ScottTodd) [inline build_tools/cmake/CMakeLists.txt:62]: These CMake files are also private: https://github.com/iree-org/iree/blob/main/build_tools/cmake/README.md

If the project is complex enough that out of tree usage feels the need to use these CMake helpers, _that's_ what needs to be fixed, _not_ making the helper functions public. Downstream users of the C/C++ code should be able to use any build system they want, not just CMake.

Comment (makslevental) [reply build_tools/cmake/CMakeLists.txt:62]: The goal here was to be able to do `iree-install-dist` and have enough to build tools (such as `iree-opt`): https://github.com/makslevental/iree-standalone

**Pull request closed without merging**